packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -171,21 +171,23 @@ - (CGAffineTransform)fixTransform:(AVAssetTrack*)videoTrack {
   CGAffineTransform transform = videoTrack.preferredTransform;
   // TODO(@recastrodiaz): why do we need to do this? Why is the preferredTransform incorrect?
   // At least 2 user videos show a black screen when in portrait mode if we directly use the
-  // videoTrack.preferredTransform Setting tx to the height of the video instead of 0, properly
-  // displays the video https://github.com/flutter/flutter/issues/17606#issuecomment-413473181
-  if (transform.tx == 0 && transform.ty == 0) {
-    NSInteger rotationDegrees = (NSInteger)round(radiansToDegrees(atan2(transform.b, transform.a)));
-    NSLog(@"TX and TY are 0. Rotation: %ld. Natural width,height: %f, %f", (long)rotationDegrees,
-          videoTrack.naturalSize.width, videoTrack.naturalSize.height);
-    if (rotationDegrees == 90) {
-      NSLog(@"Setting transform tx");
-      transform.tx = videoTrack.naturalSize.height;
-      transform.ty = 0;
-    } else if (rotationDegrees == 270) {
-      NSLog(@"Setting transform ty");
-      transform.tx = 0;
-      transform.ty = videoTrack.naturalSize.width;
-    }
+  // videoTrack.preferredTransform. This is because the transform.tx and transform.ty values are
+  // incorrectly set to 0.
+  // Setting the transform.tx value to the height of the video instead of 0 when rotationDegrees ==
+  // 90 and transform.ty to the video width when rotationDegrees == 270, properly displays the video
+  // https://github.com/flutter/flutter/issues/17606#issuecomment-413473181 In 1 other user video
+  // the transform.x and transform.y are set to 1080.0 and 0.0 respectively, whilst the width,
+  // height and rotation of the video are 848.0, 480.0 and 90 respectively. Replacing the value of
+  // transform.tx to the video height properly renders the video.
+  NSInteger rotationDegrees = (NSInteger)round(radiansToDegrees(atan2(transform.b, transform.a)));
+  NSLog(@"VIDEO__ %f, %f, %f, %f, %li", transform.tx, transform.ty, videoTrack.naturalSize.height,
```

Comment:
shouldn't this be removed or hidden behind a debug flag?

Reply:
What would you suggest?

Normally I'd use `os_log` (iOS 10+) but given the iOS 9 minimum at the moment don't think we can here? Also there doesn't seem to be any standardised approach to native logging in any of the other plugins that I could see (a few NSLogs littered around) - perhaps this is something that could be addressed in separate PR? This log was here previously prior to this change albeit different wording so I didn't see any harm in it

Reply:
I agree - if there are NSLog statements already then a new PR makes more sense indeed. 

NSLog itself is fine IMO, it's just the fact that you'll get these messages during debugging (in Xcode only I guess) with no way to deactivate these.